### PR TITLE
Bbbswa 44

### PR DESCRIPTION
**Pull request by BenCook0**
API talks to firebase correctly

### CODE REVIEW

File: backEnd/tasks/taskFactory.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  ["BIG Supporters - Family/Partner", 30],
+  ["BIG Supporters - Personal", 30],
+  ["BIG Supporters - Employer", 30],
+  ["BIG Fundamentals", 60],
+  ["BIG Extras - Car Insurance", 60],
+  ["BIG Bio", 60]
+])
+
+//date format
+dateFormat.i18n = {
```
Comment (ckthiessen) [inline backEnd/tasks/taskFactory.js:17]: Why did you choose to go full date and time for the format? Also a lot of this is already built in to the JS Date() library. Look into toLocaleDateString() and toLocaleString(). They have an options param for like every date format. We should probably just use that instead. 
https://medium.com/swlh/use-tolocaledatestring-to-format-javascript-dates-2959108ea020
(Open in incognito tab if this link is locked behind a paywall)

File: backEnd/index.js [hunk 2]
```diff
@@ -42,114 +16,106 @@ app.listen(port, () => {
...
+    toCreate.isAdmin = false;
+    toCreate.isCommunityMentor = false;
+    toCreate.requiresHomeAssessment =  false;
+      
+    await userRepository.createUser(toCreate); 
+    res.json(toCreate.id)
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
```
Comment (ckthiessen) [inline backEnd/index.js:82]: 404 if the requested user doesn't exist. 400 if the request validation fails. 500 if the server fails

File: backEnd/tasks/taskFactory.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+//this class builds and generates the default tasks when creating a user
```
Comment (ckthiessen) [inline backEnd/tasks/taskFactory.js:1]: Good idea moving this to a factory class!

File: backEnd/index.js [hunk 4]
```diff
@@ -42,114 +16,106 @@ app.listen(port, () => {
...
-  //todo: find user in firebase and delete
-  res.json(id + " was deleted");
+  try {
+    userValidator.validateDelete(id);
+    await userRepository.deleteUser(id);
+    res.json(id + " was deleted");
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
```
Comment (ckthiessen) [inline backEnd/index.js:99]: 404 if the requested user doesn't exist. 400 if the request validation fails. 500 if the server fails

File: backEnd/index.js [hunk 5]
```diff
@@ -42,114 +16,106 @@ app.listen(port, () => {
...
-  res.json(users);
+app.get('/api/users', async (req, res) => {
+  try{
+    console.log("get all users")
+    let users = await userRepository.getAllUsers();
+    res.json(users);
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
```
Comment (ckthiessen) [inline backEnd/index.js:32]: 404 if the requested user doesn't exist. 400 if the request validation fails. 500 if the server fails

File: backEnd/index.js [hunk 6]
```diff
@@ -42,114 +16,106 @@ app.listen(port, () => {
...
+    userValidator.validateId(id);
+
+    console.log("Get user by ID: /users/:" + id);
+    let found = await userRepository.getUserById(id);
+    console.log(found);
+    res.json(found);
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
```
Comment (ckthiessen) [inline backEnd/index.js:51]: 404 if the requested user doesn't exist. 400 if the request validation fails. 500 if the server fails

File: backEnd/index.js [hunk 7]
```diff
@@ -42,114 +16,106 @@ app.listen(port, () => {
...
+    userValidator.validateUser(toUpdate);
+
+    console.log("Updating user");
+    console.log(toUpdate);
+    await userRepository.updateUser(toUpdate);
+    res.json(toUpdate);
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
```
Comment (ckthiessen) [inline backEnd/index.js:118]: 404 if the requested client doesn't exist. 500 if the server can't update the requested client. Request validation should be done on the client side so there shouldn't be a malformed request (a 400 error)

File: backEnd/validations/userValidator.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+const FluentValidation = require('@kalstong/fluentvalidation');
```
Comment (ckthiessen) [inline backEnd/validations/userValidator.js:1]: Should validation be done on the client or server?

**Review by ckthiessen — CHANGES_REQUESTED** [review]
Few changes requested. Otherwise, good PR! Why is there this .Rhistory file? It keeps getting added but idk what it's for.

File: backEnd/validations/userValidator.js (hunk 8 above)
Comment (BenCook0) [reply backEnd/validations/userValidator.js:1]: both

**Pull request closed without merging**